src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -44,8 +44,27 @@ protected TelemetryConfigurationFactory()
         /// </remarks>
         public static TelemetryConfigurationFactory Instance
         {
-            get { return instance ?? (instance = new TelemetryConfigurationFactory()); }
-            set { instance = value; }
+            // Both get and set should write something to CoreEventSource so as to initialize it. This is a fix for below deadlock issue   
```

Comment:
@SergeyKanzhelev do we have any other place to make this change? Since User can create TelemetryConfiguration in his own way, we cannot totally prevent this issue. This only prevent the deadlock if user does not create own TC, instead rely on the default one which is create via TelemetryConfigurationFactory.
(And the original issue also pointed to TelemetryConfigurationFactory as one of the parties involved in the deadlock)